hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java

```diff
@@ -135,6 +137,89 @@ public void readWithTypeCoertion() {
         );
     }
 
+    @Test
+    public void readWithOneProcessor() {
+        MongoCollection<Document> collection = database.getCollection(collectionName);
+        collection.insertOne(new Document("firstName", "Luke").append("lastName", "Skywalker").append("jedi", "true"));
+
+        Pattern parallelismDagPattern = Pattern.compile("\\[localParallelism=(\\d)+]");
+        AtomicBoolean otherFound = new AtomicBoolean(false);
+        LogListener lookForProjectAndFilterStep = log -> {
+            String message = log.getLogRecord().getMessage();
```

Comment:
maybe it would be easier to get `dagJson` from `JobRecord` instead of relying on debug logs? `JobRecord` exists when the job is running.

Reply:
Would be best, but - as you said - it's only when job is running. I'm afraid it would cause flakiness